### PR TITLE
DataGrid selectedRows fix for $showHeader=false #9074

### PR DESCRIPTION
**Pull request by NekitoSP**
I just fixed issue #9074. 
Problem was due to column not initialized if `$showHeader=false`, cause column initialization code was placed in `renderHeaderCellContent`. And i moved `GridView`  JS initializer to `init();` instead of `run()` because `GridView` should be initialized  before its columns.

### CODE REVIEW

File: framework/grid/GridView.php [hunk 1]
```diff
@@ -272,6 +272,11 @@ public function init()
             $this->filterRowOptions['id'] = $this->options['id'] . '-filters';
         }
 
+        $id = $this->options['id'];
+        $options = Json::htmlEncode($this->getClientOptions());
+        $view = $this->getView();
+        $view->registerJs("jQuery('#$id').yiiGridView($options);");
+
```
Comment (SilverFire) [inline framework/grid/GridView.php]: Could you elaborate, please, why did you move this code from `run()` to init?
I'm not sure that it's a good idea.

File: framework/grid/CheckboxColumn.php [hunk 2]
```diff
@@ -81,6 +81,15 @@ public function init()
         if (substr_compare($this->name, '[]', -2, 2)) {
             $this->name .= '[]';
         }
+
+        $name = $this->grid->showHeader ? $this->getHeaderCheckBoxName() : NULL;
```
Comment (SilverFire) [inline framework/grid/CheckboxColumn.php]: `NULL` -> `null`

**Comment by SilverFire** [comment]
@NekitoSP would you like to continue work on this PR?

File: framework/grid/CheckboxColumn.php [hunk 3]
```diff
@@ -81,6 +81,15 @@ public function init()
...
+        $id = $this->grid->options['id'];
+        $options = json_encode([
```
Comment (samdark) [inline framework/grid/CheckboxColumn.php]: `Json::encode` helper should be used.

**Comment by SilverFire** [comment]
Moving to the milestone 2.0.8 because the PR is unfinished and @NekitoSP  does not respond
